_posts/2020-01-30-Age-Targeting-Part1.md

```diff
@@ -0,0 +1,185 @@
+We actually do not know, where those apps derive the age from.
+But often, they require a registration step where you would provide your year of birth.
+However, these informations are only given on by a subset of apps during the bidding process.
+To increase the available targeting audience we use the information we get on the small subset of devices to infer the age of other, unknown users.
+To achieve this, machine learning comes into place.
+
+To train a supervised machine learning for infearing the age over all the devices we need: 
+- a **reliable ground truth** datasets (in our case the age for certain devices);
+- the **features** which will go as inputs to the model.
+
+In this post, we only discuss the ground truth selection step, leaving the feature creation, the tuning, the train and the prediction of the model for the next time.
+
+**Introduction: the importance to seek the truth**<br/>
+The ground truth has to be reliable or:
+- the model will learn non realistic structures;
+- the performances of the model will decrease since the unreliable ground truth is based on arbitrary or random relations.
+
+Reliability is not a statistic quantity that you can find in the books.
+To summarize the concept in one phrase, we could state that your data are reliable as long as all your results follow an explainable behaviour without any inconsistenties.
+This is based on the fact, that we often find patterns which imply some problems in the data.
+We want to discover all of those patterns and exclude them from our dataset.
+
+We can structure the age information unreliability in two ways:
+1. consistency of a deviced as observed multiple times on the same app;
+2. consistency of a device as observed in different apps.
+
+**1) Inter app reliability**:<br/>
+In the plot below you can find some **examples of the age distribution** for six apps.
+![example_age_distribution](../images/Age-Targeting/example_age_distribution.png)
+Different apps clearly show different age distributions.
+However, it is clear that some of the distributions in the plot do not make sense.
+Why would people from 17-18 correspond to more than 95% of the total users of the app? (What happens to the 16ties? or 19ties?)
+Why would apps have the same numbers of devices for every age in a specific range?
+These must be very special apps, or the data do not correspond to truth.
+What we want to do is to exclude these apps.
+A rejection/accept statement is drown only when we have enough users observed in the app.
+In case the number of users is too low we do not consider them in the study and in the final ground truth selection.  
+This is not a big issue since the devices contained in these apps are few by definition, so we do not cut off too many data in this way.
+
+Let's describe the main pattern we discover in the data and how we get rid of them creating ad hoc variable via pyspark functions (spark allows to scale and run over the entire dataset).
```

Comment:
pattern**s**

Reply:
variable**s**

Reply:
Make the bracketed part to a brief explanation. Give some meat to the reasons why we are using Spark.